partner/partner-management-service/src/main/java/io/mosip/pms/partner/misp/service/impl/InfraProviderServiceImpl.java

```diff
@@ -274,7 +274,30 @@ public MISPLicenseResponseDto regenerateKey(String mispId) {
 				response.setLicenseKeyExpiry(newLicenseKey.getValidToDate());
 				response.setLicenseKeyStatus("Active");
 				response.setProviderId(mispId);
-				notify(MapperUtils.mapDataToPublishDto(newLicenseKey), EventType.MISP_LICENSE_UPDATED);
+				List<PartnerPolicyRequest> approvedPolicyMappedReq = partnerPolicyRequestRepository.findByPartnerId(mispId);
```

Comment:
Seems the whole logic should be corrected.
One MISP partner can have any number of License key created. MISP license key should NOT be bound to policy.
When the regenerate request is received, we check only for partner existence and its status and then generate the License Key. and also raise an event to IDA with MISP partner details and the new license key.

Reply:
Updated the code as suggested anusha